File: Java/day4/GroceryManager.java
import java.util.ArrayList;

import java.util.scanner;

public class GroceryManager {
String [] groceryList = new String[totalInitialLength]; 

public static void addItem(String items) {
System.out.print("Enter the number of grocery items: ");
Scanner input =new Scanner(System.in);
int totalInitialLength = input.nextInt();

String [] groceryList = new String[totalInitialLength]; 
//System.out.print("add item ");

System.out.print("Enter your " + groceryList.length + " values: ");
for (int i = 0; i < groceryList.length; i++) 
groceryList[i] = input.nextString();
groceryList.add(item);
int itemCount=0;
itemCount++;
System.out.println("" + item + "added to grocery list.");


public void removeItem(String item) {
if (groceryList.remove(item)) {
System.out.println( item + "' removed from list.");
} 
else {
System.out.println( item + "not found inlist.");
 }
 }



public static void ShowItem(String items) {

if (groceryList.isEmpty()) {
System.out.println("empty.add items");
        } 
else {
 System.out.println("Grocery List:");  }      
//Displaying arrays: printing array,elements
for (int i = 0; i < groceryList.length; i++) {
 System.out.print(groceryList[i] + " ");
}

//for (double e: groceryList) {
//System.out.println(e);
}
}




public static void main(String[] args) {
       
Scanner scanner = new Scanner(System.in);
while (true) {
System.out.println("Choose option:");
System.out.println("1. Add item");
System.out.println("2. Remove item");
System.out.println("3. Show available item");
System.out.println("0. Exit");
System.out.print("Enter your choice (1-0): ");

String answerSelected = scanner.nextLine();
switch (answerSelected) {
case "1":
System.out.print("Enter item to add: ");
String addItem = scanner.nextLine();
groceryManager.addItem(addItem);
break;
case "2":
System.out.print("Enter item to remove: ");
String removeItem = scanner.nextLine();
groceryManager.removeItem(removeItem);
break;
case "3":
groceryManager.showItems();
break;
case "0":
System.out.println("Goodbye");
return;
default:
System.out.println("Invalid choice. Please try again.");
}
}
}
}

//pg 456 lian
//+ArrayList()    Creates an empty list
//ArrayList<String> cityList = new ArrayList<>();
//+add(o: E): void  Appends a new element o at the end of this list. 
//+contains(o: Object): boolean   Returns true if this list contains the element o.
//+isEmpty(): boolean   Returns true if this list contains no elements.
//+lastIndexOf(o: Object): int
//(o: Object): boolean
//+size(): int
//+remove(index: int): boolean

File: python/dayfour.py/Grocery Manager.py
grocery_list = []

def add_item():
    item = input("Enter item to add: ")
    grocery_list.append(item)
    print(f{item}' added.")

def remove_item():
    item = input("Enter item to remove: ")
    if item in grocery_list:
        grocery_list.remove(item)
        print(f{item}' removed list.")
    else:
        print(f {item}' not found in the list.")

def show_items():
    if not grocery_list:
        print("grocery list is empty.")
    else:
        print(" Grocery List:")
        for idx, item in enumerate(grocery_list, start=1):
            print(f"{idx}. {item}")


print("Welcome to Grocery Manager ")

while True:
    print("\nChoose an option:")
    print("1. Add item")
    print("2. Remove item")
    print("3. Show all items")
    print("0. Exit")

    choice = input("Enter your choice (1-0): ")

    if choice == '1':
        add_item()
    elif choice == '2':
        remove_item()
    elif choice == '3':
        show_items()
    elif choice == '0':
        print("Goodbye")
        break
    else:
        print("Invalid choice.try again.")
